src/borg/compress.pyx

```diff
@@ -306,44 +381,40 @@ class Auto(CompressorBase):
 
     def _decide(self, data):
         """
-        Decides what to do with *data*. Returns (compressor, lz4_data).
+        Decides what to do with *data*. Returns (compressor, compressed_data).
 
-        *lz4_data* is the LZ4 result if *compressor* is LZ4 as well, otherwise it is None.
+        *compressed_data* is the LZ4 result unless *compressor* is NONE, in which
+        case it is the uncompressed NONE compressor result.
```

Comment:
2x NONE -> NONE_COMPRESSOR?

Reply:
Also, I guess it might be less confusing to just say:

There are 2 outcomes of this decision method:
- the data looks compressible (according to a more or less successful lz4 compression attempt), in that case we return (expensive_compressor_class, lz4_compressed_data)
- the data looks incompressible (according to a failed lz4 compression attempt), in that case we return (NONE_COMPRESSOR, none_compressed_data)

Reply:
plus:

note: we do not forbid to set the expensive_compressor to LZ4 (although it doesn't make sense).

Reply:
I'd rather go with the class name based "CNONE compressor" there rather than what happens to just be the global variable for it, matching the existing part of this comment and others talking about LZ4 and not LZ4_COMPRESSOR either, More consistent and less cumbersome to read. :smile: 
I'll clarify the doc string some more. You just proved it to be necessary: There actually is three outcomes, not two (thus the "unless compressor is NONE"). :smile:  Besides the two you desribed, the data may have been compressible with LZ4 yielding a ratio for which 0.97<ratio<1 holds true, in which case (LZ4_COMPRESSOR, lz4_compressed_data) is returned. That is one of the original cases, the case of NONE is new with my change and happens if LZ4 compression does not lead to a smaller result than NONE.

Reply:
CNONE is also fine with me. NONE is a bit too close to None though.

Reply:
ah, right, overlooked that case.